pkg/scheduler/scheduler.go

```diff
@@ -112,31 +114,55 @@ func (rs *RemoteScheduler) schedule(state cluster.State) {
 		return
 	}
 
+	//Set the status as reconciling
+	_, err = rs.inventory.UpdateStatus(&state, model.Reconciling)
+	if err != nil {
+		rs.logger.Infof("Failed to update cluster status as reconciling: %s", err)
+	}
+
 	//Reconcile CRD components first
 	for _, component := range components {
 		if rs.isCRDComponent(component.Component) {
-			rs.reconcile(component, state, schedulingID, doInstallCRD, concurrencyNotAllowed)
+			rs.reconcile(nil, component, state, schedulingID, doInstallCRD, concurrencyNotAllowed)
```

Comment:
Maybe we can rely on the Waiting-Group parameter to decide whether concurrency should be used, e.g. 

      func useConcurreny(wg *sync.WaitingGroup) bool {
          return wg != nil
      }